test/lint.ts

```diff
@@ -187,12 +187,15 @@ const main = async (
   // Find all unnecessary linting exceptions
   for (const linter of linters.linters) {
     if (linter.exceptions) {
-      const actual = new Set(linters.expectedFailures[linter.name]);
-      for (const exception of linter.exceptions)
-        if (!actual.has(exception))
+      const missingExceptions = linters.missingExpectedFailures[linter.name];
+      for (const exception in missingExceptions) {
+        if (missingExceptions[exception]) {
+          hasErrors = true;
```

Comment:
I think unnecessary exceptions should be treated as a lint error, but will be glad to revert this if needed.